### PR TITLE
fixed an issue with interpolation for service principals

### PR DESCRIPTION
**Pull request by stikkireddy**
Resolves issue #104

### CODE REVIEW

File: databricks_sync/sdk/generators/identity.py [hunk 1]
```diff
@@ -263,7 +269,6 @@ def _user_group_has_profile(self, user, role_arn):
                 return True
         except Exception:
             log.exception("Failed to check if user group has profile")
```
Comment (R7L208) [inline databricks_sync/sdk/generators/identity.py:271]:
```suggestion
        except Exception as e:
            log.exception("Failed to check if user group has profile with error {str(e)}")
```

Should probably catch the error message in exception here too.

File: databricks_sync/sdk/terraform.py [hunk 2]
```diff
@@ -88,7 +88,7 @@ def _cmd(self, cmds, *args, **kwargs):
         # Close buffers
         try:
             p.stdout.flush()
-        except:
+        except Exception:
             pass
         p.stdout.close()
```
Comment (R7L208) [inline databricks_sync/sdk/terraform.py:93]:
```suggestion
        except Exception as e:
            print(str(e))
        finally:
            p.stdout.close()
```

**Review by R7L208 — COMMENTED** [review]
Not Itai 😉 but noticed a few things

File: databricks_sync/sdk/terraform.py (hunk 2 above)
Comment (stikkireddy) [reply databricks_sync/sdk/terraform.py:93]: @R7L208 can you switch the print statement to log.debug?

Comment (R7L208) [reply databricks_sync/sdk/terraform.py:93]: Updated @stikkireddy

**Review by R7L208 — APPROVED** [review]
LGTM!

**Pull request merged**